webxr/headless/mod.rs

```diff
@@ -185,13 +207,34 @@ impl DeviceAPI<Surface> for HeadlessDevice {
         thread::sleep(std::time::Duration::from_millis(20));
         let mut data = self.data.lock().unwrap();
         let mut frame = data.get_frame();
+        let events = self.hit_tests.commit_tests();
+        frame.events = events;
         if data.needs_view_update {
             data.needs_view_update = false;
             frame
                 .events
                 .push(FrameUpdateEvent::UpdateViews(self.views()))
         };
 
+        if let Some(ref world) = data.world {
+            for source in self.hit_tests.tests() {
+                let ray = data.native_ray(source.ray, source.space);
+                let ray = if let Some(ray) = ray { ray } else { break };
+
+                let hits = world
+                    .regions
+                    .iter()
+                    .filter(|region| source.types.is_type(region.ty))
+                    .flat_map(|region| &region.faces)
+                    .filter_map(|triangle| triangle.intersect(ray))
+                    .map(|space| HitTestResult {
+                        space,
+                        id: source.id,
+                    });
+                frame.hit_test_results.extend(hits);
+            }
+        }
+
```

Comment:
Hit tests are performed synchronously every frame? For triangles hit-testing is cheap, but is this going to be expensive for other hit tests, e.g. meshes?

Reply:
This is just the mock backend, we don't care. In openxr we'll mostly be passing it down to the system.